src/box/errcode.h

```diff
@@ -115,286 +115,286 @@ struct errcode_record {
 
 #define ERROR_CODES(_) \
 	_(ER_UNKNOWN, 0,			"Unknown error") \
-	_(ER_ILLEGAL_PARAMS, 1,			"Illegal parameters, %s") \
-	_(ER_MEMORY_ISSUE, 2,			"Failed to allocate %u bytes in %s for %s") \
-	_(ER_TUPLE_FOUND, 3,			"Duplicate key exists in unique index \"%s\" in space \"%s\" with old tuple - %s and new tuple - %s") \
-	_(ER_TUPLE_NOT_FOUND, 4,		"Tuple doesn't exist in index '%s' in space '%s'") \
+	_(ER_ILLEGAL_PARAMS, 1,			"Illegal parameters", "details", STRING) \
+	_(ER_MEMORY_ISSUE, 2,			"Failed to allocate memory") \
+	_(ER_TUPLE_FOUND, 3,			"Duplicate key exists", "index", STRING, "space", STRING, "old_tuple", TUPLE, "new_tuple", TUPLE) \
+	/* ER_TUPLE_NOT_FOUND, 4, Unused */ \
 	_(ER_UNSUPPORTED, 5,			"%s does not support %s") \
-	_(ER_NONMASTER, 6,			"Can't modify data on a replication slave. My master is: %s") \
+	/* ER_NONMASTER, 6, Unused */ \
 	_(ER_READONLY, 7,			"Can't modify data on a read-only instance") \
-	_(ER_INJECTION, 8,			"Error injection '%s'") \
-	_(ER_CREATE_SPACE, 9,			"Failed to create space '%s': %s") \
-	_(ER_SPACE_EXISTS, 10,			"Space '%s' already exists") \
-	_(ER_DROP_SPACE, 11,			"Can't drop space '%s': %s") \
-	_(ER_ALTER_SPACE, 12,			"Can't modify space '%s': %s") \
-	_(ER_INDEX_TYPE, 13,			"Unsupported index type supplied for index '%s' in space '%s'") \
-	_(ER_MODIFY_INDEX, 14,			"Can't create or modify index '%s' in space '%s': %s") \
-	_(ER_LAST_DROP, 15,			"Can't drop the primary key in a system space, space '%s'") \
-	_(ER_TUPLE_FORMAT_LIMIT, 16,		"Tuple format limit reached: %u") \
-	_(ER_DROP_PRIMARY_KEY, 17,		"Can't drop primary key in space '%s' while secondary keys exist") \
-	_(ER_KEY_PART_TYPE, 18,			"Supplied key type of part %u does not match index part type: expected %s") \
-	_(ER_EXACT_MATCH, 19,			"Invalid key part count in an exact match (expected %u, got %u)") \
-	_(ER_INVALID_MSGPACK, 20,		"Invalid MsgPack - %s") \
-	_(ER_PROC_RET, 21,			"msgpack.encode: can not encode Lua type '%s'") \
+	_(ER_INJECTION, 8,			"Error injection", "injection_place", STRING) \
+	_(ER_CREATE_SPACE, 9,			"Failed to create space", "space", STRING, "reason", STRING) \
+	_(ER_SPACE_EXISTS, 10,			"Space already exists", "space", STRING) \
+	_(ER_DROP_SPACE, 11,			"Can't drop space", "space", STRING, "reason", STRING) \
+	_(ER_ALTER_SPACE, 12,			"Can't modify space", "space", STRING, "reason", STRING) \
+	_(ER_INDEX_TYPE, 13,			"Unsupported index type", "index", STRING, "space", STRING) \
+	_(ER_MODIFY_INDEX, 14,			"Can't create or modify index", "index", STRING, "space", STRING, "reason", STRING) \
+	_(ER_LAST_DROP, 15,			"Can't drop the primary key in a system space", "space", STRING) \
+	_(ER_TUPLE_FORMAT_LIMIT, 16,		"Tuple format limit reached", "limit", UINT) \
+	_(ER_DROP_PRIMARY_KEY, 17,		"Can't drop primary key while secondary keys exist", "space", STRING) \
+	_(ER_KEY_PART_TYPE, 18,			"Supplied key field type does not match index part type", "fieldno", UINT, "part_type", STRING) \
```

Comment:
Probably, better use "expected" instead of "part_type".

Reply:
Done.